app/src/test/webapp/app/error/error.ctrl.test.js

```diff
@@ -178,4 +190,46 @@ describe("ErrorCtrl", function() {
         });
     });
   });
+
+  describe("after calling resubmit", function() {
+    it("should call resubmitMessage() in service with a resubmittable message", function() {
+      spyOn(msgSvc, 'resubmitMessage').and.callFake(function() {
+        return $q.when({
+          data: {
+            status: "Success"
+          }
+        })
+      });
+
+      $controller("ErrorCtrl", {
+        $scope: $scope
+      });
+
+      $scope.message = resubmittableMessage;
+
+      $scope.resubmitMessage();
+
+      expect(msgSvc.resubmitMessage).toHaveBeenCalledWith(resubmittableMessage);
+    });
+
+    it("should not call resubmitMessage() in service if message is not resubmittable", function(){
```

Comment:
It looks like this validation is only happening on the frontend, not the backend. I opened #160 to fix this.